applicationinsights/applicationinsights-js.d.ts

```diff
@@ -0,0 +1,488 @@
+// Type definitions for ApplicationInsights-JS v0.22.14
+// Project: https://github.com/Microsoft/ApplicationInsights-JS
+// Definitions by: Kamil Szostak <https://github.com/kamilszostak>
+// Definitions: https://github.com/DefinitelyTyped/DefinitelyTyped
+
+declare module Microsoft.ApplicationInsights.Context {
+    class Application {
+        /**
+         * The application version.
+         */
+        ver: string;
+
+        /**
+         * The application build version.
+         */
+        build: string;
+    }
+
+    class Device {
+        /**
+         * The type for the current device.
+         */
+        type: string;
+
+        /**
+         * A device unique ID.
+         */
+        id: string;
+
+        /**
+         * The device OEM for the current device.
+         */
+        oemName: string;
+
+        /**
+         * The device model for the current device.
+         */
+        model: string;
+
+        /**
+         * The IANA interface type for the internet connected network adapter.
+         */
+        network: number;
+
+        /**
+         * The application screen resolution.
+         */
+        resolution: string;
+
+        /**
+         * The current display language of the operating system.
+         */
+        locale: string;
+
+        /**
+         * The IP address.
+         */
+        ip: string;
+
+        /**
+         * The device language.
+         */
+        language: string;
+
+        /**
+         * The OS name.
+         */
+        os: string;
```

Comment:
nitpicks: I would spell out "operatingSystem" and/or add "Name" on the end for clarity.